src/query/query.ts

```diff
@@ -0,0 +1,226 @@
+import { BaseQuery } from './base-query';
+import {
+  IConditionExpr,
+  IIndexOnParams,
+  IIndexWithParams,
+  ILetExpr,
+  IndexType,
+  ISelectType,
+  LogicalWhereExpr,
+  SortType,
+} from './interface/query-types';
+import {
+  IndexParamsOnExceptions,
+  IndexQueryException,
+  MultipleQueryTypesException,
+  SelectClauseException,
```

Comment:
`SelectClauseException` not used. I understand leaving it here if it will be used eventually?